tests/infer/mcmc/test_mcmc.py

```diff
@@ -51,6 +55,20 @@ def test_mcmc_interface():
     assert_equal(sample_std, torch.tensor([1.0]), prec=0.08)
 
 
+@pytest.mark.parametrize("num_chains", [
+    1,
+    pytest.param(2, marks=[pytest.mark.skipif("CI" in os.environ, reason="CI only provides 1 CPU"),
+                           pytest.mark.skip(reason="https://github.com/uber/pyro/issues/1699")])
```

Comment:
Let us not skip this for local testing. Alternatively, we can also capture the UserWarning if "CI" is in the environment.

Reply:
In my system, the queue stuck after 18 steps (there are totally 20 steps: 10 for warmups and 10 for samples), and there is no error raised. I have to mark this `skip` to not affect other people. I think we can remove it later when #1699 is resolved. It can be that the problem just happens in my system (despite that I have restarted it).

Reply:
Thanks for explaining. I think we should resolve this separately. For some reason, I am getting an assertion error even with `num_chains = 1`

```
>       assert diagnostics["y"]["n_eff"].shape == torch.Size([1])
E       assert torch.Size([]) == torch.Size([1])
E         Right contains more items, first extra item: 1
E         Full diff:
E         - torch.Size([])
E         + torch.Size([1])
E         ?             +

tests/infer/mcmc/test_mcmc.py:68: AssertionError
```

Reply:
Okay, I see the issue. When MKL isn't available, we return `torch.tensor(float('nan'))`. Let us change that to return the correctly shaped NaN tensor.